workflow/controller/workflowpod.go

```diff
@@ -140,7 +140,7 @@ func (woc *wfOperationCtx) createWorkflowPod(nodeName string, mainCtr apiv1.Cont
 	} else {
 		wfActiveDeadlineSeconds := int64((*wfDeadline).Sub(time.Now().UTC()).Seconds())
 		if wfActiveDeadlineSeconds <= 0 {
-			return nil, nil
+			return nil, ErrDeadlineExceeded
```

Comment:
`ErrDeadlineExceeded` means that the operation deadline was exceeded (the 10 second deadline per operation). This is overloading its definition.

Reply:
yes, you are right. we don't want to overload it